R/site-calade.R

```diff
@@ -0,0 +1,230 @@
+calade_patch_rmd <- function(path) {
+  lines <- brio::read_lines(path)
+  lines <- c(lines[1],
+             "output: hugodown::md_document",
+             lines[-1]
+  )
+  brio::write_lines(lines, path)
+}
+
+
+# Patch all rmd files in a folder
+calade_patch_rmd_dir <- function(path) {
+  rmd_files <- dir_ls(path = path, glob = "*.Rmd", recurse = TRUE)
+  lapply(rmd_files, calade_patch_rmd)
+}
+
+
+# Writes the hugodown yaml file
+calade_write_hugodown <- function(path) {
+  opts <- list(hugo_version = "0.66.0")
+  yaml::write_yaml(opts, path(path, "_hugodown.yaml"))
+}
+
+
+# Writes the readme file
+calade_write_readme <- function(path) {
+  file_copy(path_package("hugodown", "calade", "README.md"), path)
+}
+
+
+# Writes the sentinel file
+calade_write_sentinel <- function(path) {
+  file_copy(path_package("hugodown", "calade", "index.Rmd"), path)
+}
+
+
+# Copies the highlight.css style file across
+calade_write_css <- function(path) {
+  dir_create(path(path, "static", "css"))
+  file_copy(
+    path = path_package("hugodown", "calade", "highlight.css"),
+    new_path = path(path, "static", "css")
+  )
+}
+
+
+# Inserts link to highlight.css file in head_custom.html
+calade_patch_head_custom <- function(path) {
+
+  # (is this necessary?)
+  head <- path(path, "layouts", "partials", "head_custom.html")
+  dir_create(path_dir(head))
+
+  # append to file
+  lines <- brio::read_lines(head)
+  brio::write_lines(c(
+    lines,
+    "",
+    "<!-- css for syntax highlighting -->",
+    "<link rel='stylesheet' href='{{ \"css/highlight.css\" | relURL }}' title='hl'>",
+    "{{ range .Params.html_dependencies }}",
+    "  {{ . | safeHTML }}",
+    "{{ end }}"
+  ), head)
+}
+
+
+# Patches the config.toml file for the example site. Specifically, the
+# config must allow the markdown renderer to pass raw html. Also needs to
+# specify the publishDir
+calade_patch_config <- function(path) {
+
+  config <- path(path, "config.toml")
+  lines <- brio::read_lines(config)
+
+  # append to existing
+  brio::write_lines(c(
+    lines,
+    '',
+    '',
+    '# A hugodown site requires that Hugo be explicitly',
+    '# told how to handle markup. Because hugodown generates',
+    '# the raw HTML for R code chunks, the "unsafe = true"',
+    '# setting is required, or else Hugo will not allow the',
+    '# raw HTML to be passed from the .md file to the .html',
+    '# file. See:',
+    '# https://gohugo.io/getting-started/configuration-markup',
+    '[markup]',
+    '  defaultMarkdownHandler = "goldmark"',
+    '  [markup.goldmark]',
+    '    [markup.goldmark.renderer]',
+    '      unsafe = true',
+    ''
+  ), config)
+}
+
+
+# Throws error if pandoc is not installed, and warns if pandoc version
+# is too low (needed to ensure calade example site pages knit to md)
+pandoc_check <- function(version = NULL) {
+
+  # stop if no pandoc
+  if(!rmarkdown::pandoc_available()) {
+    stop("Could not find a pandoc installation", call. = FALSE)
```

Comment:
Can you please use `abort()` here? (It does `call. = FALSE` by default and gives better tracebacks)